extensions/2.0/Vendor/EXT_meshopt_compression/README.md

```diff
@@ -191,31 +193,52 @@ The encoded stream structure is as follows:
 - One or more attribute blocks, detailed below
 - Tail block, which consists of a baseline element stored verbatim, padded to 32 bytes
 
-Each attribute block stores a sequence of deltas, with the first element in the first block using the deltas from the baseline element stored in the tail block, and each subsequent element using the deltas from the previous element. The attribute block always stores an integer number of elements, with that number computed as follows:
+Note that there is no way to calculate the length of a stream; instead, it is expected that the input stream is correctly sized (using `byteLength`) so that the tail block element can be found.
+
+Each attribute block stores a sequence of deltas, with the first element in the first block using the deltas from the baseline element stored in the tail block, and each subsequent element using the deltas from the previous element. The attribute block always stores up to an integer number of elements, with that number computed as follows:
 
 ```
-blockSize = min((8192 / byteStride) & ~15, 256)
+maxBlockElements = min((8192 / byteStride) & ~15, 256)
 ```
 
-The attribute block structure consists of `byteStride` blocks (one for each byte of the element) with the following structure:
+Each attribute block consists of `byteStride` "data blocks" (one for each byte of the element), and each "data block" contains deltas stored for groups of elements. Each group always contains 16 elements; when the number of elements that needs to be encoded isn't divisible by 16, it gets rounded up and the remaining elements are ignored after decoding. In other terms:
 
-- Header bits, 2 bits for each group of 16 elements (`blockSize`/16 2-bit values), padded to a byte
-- Delta blocks, with variable number of bytes stored for each group of 16 elements
+```
+groupCount = ceil(blockElements / 16)

```

Comment:
Clarify where blockElements comes from. I did it as follows in my implementation, but I would prefer it to be spelled out, especially if this is not correct.

> ```
> blockElements = min(remainingBlockElements, maxBlockElements)
> ```
> 
> where `remainingBlockElements` is the number of elements that have yet to be written. A stream with a byteStride of 64 containing 200 elements would be broken up into two attribute blocks: one containing 128 elements, and the other containing 72 elements. And these blocks would have 8 and 5 groups, respectively.

Reply:
Yeah this is correct. I've removed "up to" from the description of the element count and changed `remainingBlockElements` to simply `remainingElements` to be more clear, but otherwise incorporated this into the spec edits.